subsys/emds/emds.c

```diff
@@ -11,60 +11,68 @@
 #include <drivers/flash.h>
 #include "emds_flash.h"
 
-#if defined(CONFIG_BT) && !defined(CONFIG_BT_LL_SW_SPLIT)
-#include <sdc.h>
-#include <mpsl.h>
-#endif
-
 #include <logging/log.h>
 LOG_MODULE_REGISTER(emds, CONFIG_EMDS_LOG_LEVEL);
 
 K_SEM_DEFINE(emds_sem, 0, 1);
 static bool emds_ready;
 static bool emds_initialized;
 static bool emds_complete;
```

Comment:
Then we have to remove the function about the check of completion.

Reply:
Will this function work well if there thread with the infinite loop? What happens if a customer calls this function when one loop passes but not all data are stored yet? Is it possible?

IMO callback is much better rather than the polling function. I'd suggest to remove the polling function at all and to rely on callback.